mkosi/config.py

```diff
@@ -170,32 +732,34 @@ def parse(self, path: Path, namespace: argparse.Namespace) -> argparse.Namespace
 
         if "Match" in parser.sections():
             for k, v in parser.items("Match"):
-                if not (s := self.lookup.get(k)):
-                    die(f"Unknown setting {k}")
+                if (s := self.settings_lookup.get(k)):
+                    if not (match := s.match):
+                        die(f"{k} cannot be used in [Match]")
 
-                if not (match := s.match):
-                    die(f"{k} cannot be used in [Match]")
+                    # If we encounter a setting in [Match] that has not been explicitly configured yet, we assign
+                    # it it's default value first so that we can [Match] on default values for settings.
```

Comment:
```suggestion
                    # it its default value first so that we can [Match] on default values for settings.
```
ah, homophone errors... happen to me all the time, too.